File: java-selenium/src/main/java/com/orangehrm/pages/PIMPage.java
package com.orangehrm.pages;

import com.orangehrm.base.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class PIMPage extends BasePage {

    @FindBy(xpath = "//h6[text()='PIM']")
    private WebElement pimPageTitle;

    @FindBy(xpath = "//button[normalize-space()='Add']")
    private WebElement addEmployeeButton;

    @FindBy(xpath = "//input[@placeholder='Type for hints...']")
    private WebElement employeeNameSearchField;

    @FindBy(xpath = "//button[@type='submit']")
    private WebElement searchButton;

    public PIMPage(WebDriver driver) {
        super(driver);
    }

    public boolean isPIMPageDisplayed() {
        return isDisplayed(pimPageTitle);
    }

    public String getPIMPageTitle() {
        return getText(pimPageTitle);
    }

    public void clickAddEmployeeButton() {
        click(addEmployeeButton);
    }

    public void searchEmployee(String employeeName) {
        sendKeys(employeeNameSearchField, employeeName);
        click(searchButton);
    }
}

File: playwright-typescript/src/pages/pim-page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from '../base/base-page';

export class PIMPage extends BasePage {
  private readonly pimPageTitle: Locator;
  private readonly addEmployeeButton: Locator;
  private readonly employeeNameSearchField: Locator;
  private readonly searchButton: Locator;

  constructor(page: Page) {
    super(page);
    this.pimPageTitle = page.locator('h6:has-text("PIM")');
    this.addEmployeeButton = page.locator('button:has-text("Add")');
    this.employeeNameSearchField = page.getByPlaceholder('Type for hints...');
    this.searchButton = page.locator('button[type="submit"]');
  }

  async isPIMPageDisplayed(): Promise<boolean> {
    return await this.isDisplayed(this.pimPageTitle);
  }

  async getPIMPageTitle(): Promise<string> {
    return await this.getText(this.pimPageTitle);
  }

  async clickAddEmployeeButton(): Promise<void> {
    await this.click(this.addEmployeeButton);
  }

  async searchEmployee(employeeName: string): Promise<void> {
    await this.fill(this.employeeNameSearchField, employeeName);
    await this.click(this.searchButton);
  }
}


